File: src/entity/Produit.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entity;

import java.sql.Date;
import java.time.LocalDate;

/**
 *
 * @author Brahim
 */
public class Produit {
    private int IdProduit;
    private String NomProduit;
    private String DescProduit;
    private LocalDate DateProduit;
    private float PrixProduit ;
    private int Qte;
    private String PhotoP;
    private String NomCategorie;

    public Produit(int IdProduit, String NomProduit) {
        this.IdProduit = IdProduit;
        this.NomProduit = NomProduit;
    }

    public Produit(LocalDate DateProduit, String NomProduit, String DescProduit, int Qte, float PrixProduit, String NomCategorie, String PhotoP) {
     
        this(Qte, NomProduit, DescProduit, DateProduit, PrixProduit, PhotoP);
          
    }

    public Produit(int Qte, String NomProduit, String DescProduit, LocalDate DateProduit, float PrixProduit, String PhotoP) {
        this.NomProduit = NomProduit;
        this.DescProduit = DescProduit;
        this.DateProduit = DateProduit;
        this.PrixProduit = PrixProduit;
        this.Qte = Qte;
        this.PhotoP = PhotoP;
        this.NomCategorie = NomCategorie;
    }

    public Produit(int IdProduit, String NomProduit, String DescProduit, LocalDate DateProduit, float PrixProduit, int Qte, String PhotoP) {
        this.IdProduit = IdProduit;
        this.NomProduit = NomProduit;
        this.DescProduit = DescProduit;
        this.DateProduit = DateProduit;
        this.PrixProduit = PrixProduit;
        this.Qte = Qte;
        this.PhotoP = PhotoP;
    }

    public Produit(int IdProduit , String NomProduit, String DescProduit, LocalDate DateProduit, float PrixProduit,int Qte, String PhotoP,String NomCategorie) {
        
        this.IdProduit = IdProduit;
        this.NomProduit = NomProduit;
        this.DescProduit = DescProduit;
        this.DateProduit = DateProduit;
        this.PrixProduit = PrixProduit;
        this.Qte = Qte;
        this.PhotoP = PhotoP;
        this.NomCategorie = NomCategorie;
    }

    public Produit() {
    }

    public int getIdProduit() {
        return IdProduit;
    }

    public void setIdProduit(int IdProduit) {
        this.IdProduit = IdProduit;
    }

    public String getNomProduit() {
        return NomProduit;
    }

    public void setNomProduit(String NomProduit) {
        this.NomProduit = NomProduit;
    }

    public String getDescProduit() {
        return DescProduit;
    }

    public void setDescProduit(String DescProduit) {
        this.DescProduit = DescProduit;
    }

    public LocalDate getDateProduit() {
        return DateProduit;
    }

    public void setDateProduit(LocalDate DateProduit) {
        this.DateProduit = DateProduit;
    }

    public float getPrixProduit() {
        return PrixProduit;
    }

    public void setPrixProduit(float PrixProduit) {
        this.PrixProduit = PrixProduit;
    }

    public int getQte() {
        return Qte;
    }

    public void setQte(int Qte) {
        this.Qte = Qte;
    }

    public String getPhotoP() {
        return PhotoP;
    }

    public void setPhotoP(String PhotoP) {
        this.PhotoP = PhotoP;
    }
    
    public String getNomCategorie() {
        return NomCategorie;
    }

    public void setNomCategorie(String NomCategorie) {
        this.NomCategorie = NomCategorie;
    }

    @Override
   
    public String toString() {
        return NomProduit;
    }

    
    
    
}

File: Downloads/integration web/src/Entity/Produit.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

/**
 * Produit
 *
 * @ORM\Table(name="produit", indexes={@ORM\Index(name="IdCategorie", columns={"IdCategorie"})})
 * @ORM\Entity
 */
class Produit
{
    /**
     * @var int
     *
     * @ORM\Column(name="IdProduit", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idproduit;

    /**
     * @var string
     *
     * @ORM\Column(name="NomProduit", type="string", length=255, nullable=false)
     */
    private $nomproduit;

    /**
     * @var string
     *
     * @ORM\Column(name="DescProduit", type="string", length=255, nullable=false)
     */
    private $descproduit;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="DateProduit", type="date", nullable=false)
     */
    private $dateproduit;

    /**
     * @var float
     *
     * @ORM\Column(name="PrixProduit", type="float", precision=255, scale=0, nullable=false)
     */
    private $prixproduit;

    /**
     * @var int
     *
     * @ORM\Column(name="Qte", type="integer", nullable=false)
     */
    private $qte;

    /**
     * @var string
     *
     * @ORM\Column(name="PhotoP", type="string", length=255, nullable=false)
     */
    private $photop;

    /**
     * @var \Categorie
     *
     * @ORM\ManyToOne(targetEntity="Categorie")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="IdCategorie", referencedColumnName="IdCategorie")
     * })
     */
    private $idcategorie;

    public function getIdproduit(): ?int
    {
        return $this->idproduit;
    }

    public function getNomproduit(): ?string
    {
        return $this->nomproduit;
    }

    public function setNomproduit(string $nomproduit): self
    {
        $this->nomproduit = $nomproduit;

        return $this;
    }

    public function getDescproduit(): ?string
    {
        return $this->descproduit;
    }

    public function setDescproduit(string $descproduit): self
    {
        $this->descproduit = $descproduit;

        return $this;
    }

    public function getDateproduit(): ?\DateTimeInterface
    {
        return $this->dateproduit;
    }

    public function setDateproduit(\DateTimeInterface $dateproduit): self
    {
        $this->dateproduit = $dateproduit;

        return $this;
    }

    public function getPrixproduit(): ?float
    {
        return $this->prixproduit;
    }

    public function setPrixproduit(float $prixproduit): self
    {
        $this->prixproduit = $prixproduit;

        return $this;
    }

    public function getQte(): ?int
    {
        return $this->qte;
    }

    public function setQte(int $qte): self
    {
        $this->qte = $qte;

        return $this;
    }

    public function getPhotop(): ?string
    {
        return $this->photop;
    }

    public function setPhotop(string $photop): self
    {
        $this->photop = $photop;

        return $this;
    }

    public function getIdcategorie(): ?Categorie
    {
        return $this->idcategorie;
    }

    public function setIdcategorie(?Categorie $idcategorie): self
    {
        $this->idcategorie = $idcategorie;

        return $this;
    }


}
